### PR TITLE
MGDOBR-430: Slack Processor creation is failing with Error when calling Kafka instance TopicsApi (status=409)

### PR DESCRIPTION
**Pull request by manstis**
[MGDOBR-430](https://issues.redhat.com/browse/MGDOBR-430) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: rhoas-client/src/main/java/com/redhat/service/bridge/rhoas/RhoasClientImpl.java [hunk 1]
```diff
@@ -50,11 +50,34 @@ public RhoasClientImpl(KafkasMgmtV1Client mgmtClient, KafkaInstanceAdminClient i
...
+
+    private Uni<Topic> handleCreateTopicFailure(String topicName, Throwable f) {
+        if (!(f instanceof AppServicesException)) {
+            RhoasClientException rhoasClientException = logAndWrapFailure("Error when creating topic " + topicName, f);
+            return Uni.createFrom().failure(rhoasClientException);
+        }
+        AppServicesException exception = (AppServicesException) f;
+        if (exception.getStatusCode() == HttpStatus.SC_CONFLICT) {
+            LOG.info("Topic {} has already been created", topicName);
+            return getTopic(topicName);
```
Comment (manstis) [inline rhoas-client/src/main/java/com/redhat/service/bridge/rhoas/RhoasClientImpl.java:76]: I call our "public" `getTopic(topicName)` method to support the potential for it too to fail.

**Comment by manstis** [comment]
Well... that works well 😆 

Seems like my change needs further love.

**Comment by manstis** [comment]
This may not be an issue with my PR. See https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/446 too.

**Comment by sutaakar** [comment]
The Manager - CI issues should be addressed by https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/431

**Review by r00ta — APPROVED** [review]

**Review by lucamolteni — APPROVED** [review]

**Pull request merged**